beacon-chain/core/helpers/committee.go

```diff
@@ -31,6 +32,30 @@ type shufflingInput struct {
 	committeesPerEpoch uint64
 }
 
+// ValidatorAssignment defines the assignment information needed by
+// validators every epoch.
+type ValidatorAssignment struct {
+	Committee  []uint64
+	Shard      uint64
+	Slot       uint64
+	IsProposer bool
+}
+
+var currentEpoch uint64
```

Comment:
@IvanTheGreatDev Instead of making all these separate vars can you put this in a struct ? That way we can add a few methods and validate the cache requests coming in. Ex: if the cache request is for a previous or future epoch we can reject the request, etc. We can even cache assignments for different epochs and depending on if it is saved return them. This can further reduce computation time